coordinate/coordinator.go

```diff
@@ -0,0 +1,301 @@
+package coordinate
+
+import (
+	"errors"
+	"log"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/go-etcd/etcd"
+	"github.com/tylertreat/Flotilla/flotilla-server/daemon/broker"
+)
+
+// ErrorUnableToConnect - Was not able to connect to the etcd instance
+var ErrorUnableToConnect = errors.New("Unable to Connect to etcd")
+
+// ErrorUnableToStartCluster - can't start the cluster
+var ErrorUnableToStartCluster = errors.New("Unable to start cluster")
+
+// ErrorUnableToStopCluster - can't stop the cluster
+var ErrorUnableToStopCluster = errors.New("Unable to stop cluster")
+
+const (
+	rootDir    = "/running/"
+	config     = "/config"
+	defaultTTL = 1200
+)
+
+// WatchFunc is a type that is used to handle responses from watches that are
+// created against etcd
+type WatchFunc func(*etcd.Response)
+
+// Client wraps the go-etcd package and its interactions with etcd into simple
+// operations that the flotilla-client and flotilla-server can use to spin up a
+// test cluster. the flotilla-client manages the cluster and watches for
+// for flotilla-server instances to join the flota specific cluster.
+type Client struct {
+	client     *etcd.Client
+	flota      string
+	id         string
+	watchLock  sync.Mutex
+	watchList  map[string]WatchFunc
+	stopList   map[string]chan bool
+	Registered bool
+}
+
+// getOrCreateRoot will delete the root + flota and its children if they do exist
+func (c *Client) getOrCreateRoot() error {
+
+	resp, err := c.client.Get(rootDir+c.flota, false, false)
+	if err != nil {
+		return nil
+	}
+
+	if resp == nil {
+		_, er := c.client.CreateDir(rootDir+c.flota, defaultTTL)
+		if er != nil {
+			return er
+		}
+		log.Println("Create new run ", rootDir+c.flota)
+
+	} else {
+
+		_, der := c.client.Delete(rootDir+c.flota, true)
+		if der != nil {
+			return der
+		}
+		log.Println("Remove previous run ", rootDir+c.flota)
+
+		_, cer := c.client.CreateDir(rootDir+c.flota, defaultTTL)
+		if cer != nil {
+			return cer
+		}
+		log.Println("Create new run ", rootDir+c.flota)
+	}
```

Comment:
This can be refactored. Check if `resp != nil` and delete it if true. In either case, you always create the new run but don't have the logic duplicated.